### PR TITLE
Adds search functionality

### PR DESCRIPTION
**Pull request by Melisa-Zorraindo**
I've added functionality for the search feature. The filters will come at a later stage, as previously discussed.

The search component supports keyboard navigation to make it more accessible. Currently, clicking/pressing 'Enter' on an item from the autocomplete list doesn't take the user to the venue they've chosen. I will start working on that page now and make a different PR for it.

The component is also hidden from view on small screens. Since the navigation bar will also be affected on small screens, I'll gather these two features in their own PR.

Aside from the components created, I also made an API hook since there are now two pages that make API calls.

New components:
- Feedback
- Search
- Sidebar (children: Search and Filter components in the future)
- SidebarHeader

The folder 'types' still exists because I'm using the CardProps interface in various places throughout the app.

### CODE REVIEW

File: src/components/Search/index.tsx [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+
+    const selectedSuggestion = suggestions[selected].id;
+    suggestions.forEach((suggestion) => {
+      const li = document.getElementById(suggestion.id);
+
+      suggestion.id === selectedSuggestion
+        ? li?.classList.add(styles.selected)
+        : li?.classList.remove(styles.selected);
+
+      //The number 810 comes from logging the offset to the console and selecting a number between the 4th item (807) and the 5th (865)
```
Comment (chrislewispac) [inline src/components/Search/index.tsx:76]: this isn't going to be the best way to do this i think. :) Let's figure out a way without magic numbers.

File: src/hooks/apiHook.tsx [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  const [data, setData] = useState<CardProps[]>([]);
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+  const [isError, setIsError] = useState<boolean>(false);
+
+  useEffect(() => {
+    async function fetchData(): Promise<void> {
+      try {
+        setIsLoading(true);
+        const response = await fetch(url);
+        const json = await response.json();
```
Comment (chrislewispac) [inline src/hooks/apiHook.tsx:18]: does this need to be awaited?

**Review by chrislewispac — APPROVED** [review]

**Comment by chrislewispac** [comment]
try to do really small PRs, when you go to work professionally you will hear this a LOT

File: src/components/Search/index.tsx (hunk 1 above)
Comment (Melisa-Zorraindo) [reply src/components/Search/index.tsx]: Thanks for this! That block was unnecessarily complicated, I didn't need to check for the element's position at all

File: src/hooks/apiHook.tsx (hunk 2 above)
Comment (Melisa-Zorraindo) [reply src/hooks/apiHook.tsx:18]: I get a promise pending as a response if I don't await it

**Pull request merged**